src/util.jl

```diff
@@ -40,22 +40,20 @@ end
 
 # Something doesn't match up type wise
 function check_array_parameters(Size, T, N, L)
-    (!isa(Size, DataType) || (Size.name !== Tuple.name)) && throw(ArgumentError("Static Array parameter Size must be a Tuple type, got $Size"))
-    !isa(T, Type) && throw(ArgumentError("Static Array parameter T must be a type, got $T"))
-    !isa(N.parameters[1], Int) && throw(ArgumenError("Static Array parameter N must be an integer, got $(N.parameters[1])"))
-    !isa(L.parameters[1], Int) && throw(ArgumentError("Static Array parameter L must be an integer, got $(L.parameters[1])"))
+    (!isa(Size, DataType) || (Size.name !== Tuple.name)) && return throw(ArgumentError("Static Array parameter Size must be a Tuple type, got $Size"))
+    !isa(T, Type) && return throw(ArgumentError("Static Array parameter T must be a type, got $T"))
+    !isa(N.parameters[1], Int) && return throw(ArgumenError("Static Array parameter N must be an integer, got $(N.parameters[1])"))
+    !isa(L.parameters[1], Int) && return throw(ArgumentError("Static Array parameter L must be an integer, got $(L.parameters[1])"))
     # shouldn't reach here. Anything else should have made it to the function below
-    error("Internal error. Please file a bug")
+    return error("Internal error. Please file a bug")
 end
 
-@generated function check_array_parameters{Size,T,N,L}(::Type{Size}, ::Type{T}, ::Type{Val{N}}, ::Type{Val{L}})
+function check_array_parameters{Size,T,N,L}(::Type{Size}, ::Type{T}, ::Type{Val{N}}, ::Type{Val{L}})
```

Comment:
This is the version of `check_array_parameters()` which will be called, and I think there was a reason to have it as a `@generated` function - it's fairly important that this does compile down to nothing in the case that the checks pass.

Did you check that this actually compiles down to nothing, even when it's not a generated function? I'd suggest you just leave this as it was, but include the fix you had before, to throw in the generated code, rather than in the generator itself.